### ISSUE TITLE
Add class to my best.pt

### ISSUE THREAD

**Issue by biscowar** [ultralytics/ultralytics#3304]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have trained my model for car make and model recognition. The result is my best.pt.
The model working fine. Now i try to add more class to my Yaml file example

add 599: Porsche_Panamera to yaml
add 500 jpg and labels to train directory
add 50 jpg and labels to valid and to test directory

I run the training with this code
 
from ultralytics import YOLO
             
# Load the model.
model = YOLO('best.pt')
 
# Training.
if __name__ == "__main__":
	results = model.train(
   	data='add.yaml',
   	imgsz=640,
   	epochs=10,
   	batch=48,
   	name='make_modelnew'
	)

The problem i that my new best.pt model recognize only the prosche and forget all the old model i trained before. Look like i have to train the whole dataset every time i have a new model to add

**Comment by Kayzwer** [comment]
you can train your model on new data set by using previous .pt but only the parameters of head will not being transfer

**Comment by pderrenger** [comment]
@Kayzwer hey! It's great to hear that your model is working fine. When you add a new class to your YAML file and retrain the model with the new dataset, the new best.pt model might prioritize the new class and seem to forget the old classes. This happens because when you retrain the model, only the parameters of the head are updated, while the body remains unchanged.

To address this, you can try training your model with the previous .pt file, and as the model trains on the new dataset, the parameters of the head will be updated to accommodate the new class. This should help in retaining the knowledge from your previous training while adapting to the new class.

I hope this helps! Let me know if you have any other questions.

**Comment by rwik04** [comment]
What is meant by previous.pt, I have only best.pt and last.pt @pderrenger

**Comment by glenn-jocher** [comment]
@rwik04, the `previous.pt` mentioned refers to any pre-trained model checkpoint you have, such as `best.pt` or `last.pt`. These files contain the weights of your model from a previous training session. You can use either of these checkpoints to continue training on new data while retaining the knowledge the model has already learned. `best.pt` typically represents the model with the best validation performance from the previous training, while `last.pt` is simply the last set of weights saved. 🎯
